openstack_controller/tests/test_openstack.py

```diff
@@ -96,6 +82,19 @@ def test_populate_servers_cache_with_project_name_none(servers_detail, aggregato
     assert 'other-2' in servers
 
 
+@mock.patch('datadog_checks.openstack_controller.api.ApiFactory.create',
+            return_value=mock.MagicMock(AbstractApi))
+def test_check(mock_api, aggregator):
+    check = OpenStackControllerCheck("test", {'ssl_verify': False}, {}, instances=INSTANCES)
+
+    check.check(INSTANCES[0])
+
+    aggregator.assert_service_check('openstack.keystone.api.up', AgentCheck.OK)
+    aggregator.assert_service_check('openstack.nova.api.up', AgentCheck.OK)
+    aggregator.assert_service_check('openstack.neutron.api.up', AgentCheck.OK)
+    mock_api.assert_called_with(ANY, ANY, INSTANCES[0])
+
+
```

Comment:
I had also added this other test, which was failing when check method still was checking for keystone_server_url but I dont think it's necessary anymore. I can add it back:
```
@mock.patch('datadog_checks.openstack_controller.api.ApiFactory.create',
            return_value=mock.MagicMock(AbstractApi))
def test_check_with_config_file(mock_api, aggregator):
    instances = copy.deepcopy(INSTANCES)
    del instances[0]['keystone_server_url']
    instances[0]['openstack_config_file_path'] = os.path.abspath('./tests/fixtures/openstack_config.yaml')
    instances[0]['openstack_cloud_name'] = 'test_cloud'
    check = OpenStackControllerCheck("test", {'ssl_verify': False}, {}, instances=instances)

    check.check(instances[0])

    aggregator.assert_service_check('openstack.keystone.api.up', AgentCheck.OK)
    aggregator.assert_service_check('openstack.nova.api.up', AgentCheck.OK)
    aggregator.assert_service_check('openstack.neutron.api.up', AgentCheck.OK)
    mock_api.assert_called_with(ANY, ANY, instances[0])
```